x/trigger/keeper/msg_server.go

```diff
@@ -0,0 +1,71 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the account MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+// CreateTrigger creates new trigger from msg
+func (s msgServer) CreateTrigger(goCtx context.Context, msg *types.MsgCreateTriggerRequest) (*types.MsgCreateTriggerResponse, error) {
```

Comment:
What prevents me from creating a trigger with a message that I shouldn't be allowed to do.

E.g. I create trigger for 10 blocks out with a message that sends all of your funds to me.

Usually this is handled because a key part of the message (e.g. the `From`) must use their private key to sign the message before sending it in. But none of that signature stuff is involved in the messages being executed by a trigger.

Reply:
This is a pretty big issue, and I will need to think about this.

Reply:
My initial thought would be to:
1. Expand the `MsgCreateTriggerRequest` to allow multiple signers.
2. As part of `ValidateBasic()`, make sure that all the `msg.GetSigners()` slices equal the `MsgCreateTriggerRequest.GetSigners()` slice.

Reply:
Just to add a little bit of clarification to this. I updated this so the action signers must be included in MsgCreateTriggerRequest.GetSigners() rather than it being an exact match.